# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Lowering.cpp
# Commit: 1e582b17eab7
# Full Hash: 1e582b17eab7abf3958e46bc52ea3eeafa764eeb
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 09:30:56
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.6: Add tls dependency for truncate i32. r=lth
#   
#   We generate builtin call for MTruncateToInt32 operation for floating points types,
#   so we need to add a tls dependency.
#   I inserted NYI for arm64 because Ion doesn't support arm64.
# ==============================================================================

diff -r a309060018a8 -r 1e582b17eab7 js/src/jit/Lowering.cpp
--- a/js/src/jit/Lowering.cpp	Wed Sep 09 06:37:59 2020 +0000
+++ b/js/src/jit/Lowering.cpp	Wed Sep 09 08:59:55 2020 +0000
@@ -2357,6 +2357,16 @@
   }
 }
 
+void LIRGenerator::visitWasmBuiltinTruncateToInt32(
+    MWasmBuiltinTruncateToInt32* truncate) {
+  MDefinition* opd = truncate->input();
+  MOZ_ASSERT(opd->type() == MIRType::Double || opd->type() == MIRType::Float32);
+
+  // May call into JS::ToInt32() on the slow OOL path.
+  gen->setNeedsStaticStackAlignment();
+  lowerWasmBuiltinTruncateToInt32(truncate);
+}
+
 void LIRGenerator::visitWasmBoxValue(MWasmBoxValue* ins) {
   LWasmBoxValue* lir = new (alloc()) LWasmBoxValue(useBox(ins->input()));
   define(lir, ins);